samples/nrf9160/modem_shell/src/ping/icmp_ping_shell.c

```diff
@@ -0,0 +1,269 @@
+
+			if (ctx_info_tbl->array[0].mtu != 0) {
+				ping_args->mtu = ctx_info_tbl->array[0].mtu;
+			} else {
+				ping_args->mtu = ICMP_DEFAULT_LINK_MTU;
+			}
+
+			found = true;
+		}
+	}
+	return found;
+}
+/*****************************************************************************/
+
+int icmp_ping_shell(const struct shell *shell, size_t argc, char **argv)
+{
+	struct icmp_ping_shell_cmd_argv ping_args;
+	int flag, dest_len;
+
+	icmp_ping_shell_cmd_defaults_set(&ping_args);
+
+	if (argc < 3) {
+		goto show_usage;
+	}
+
+	/* Start from the 1st argument */
+	optind = 1;
+
+	while ((flag = getopt_long(argc, argv, "d:t:c:i:I:l:h6", long_options,
+				   NULL)) != -1) {
+		switch (flag) {
+		case 'd': /* destination */
+			dest_len = strlen(optarg);
+			if (dest_len > ICMP_MAX_URL) {
+				shell_error(shell, "too long destination name");
+				goto show_usage;
+			}
+			strcpy(ping_args.target_name, optarg);
+			break;
+		case 't': /* timeout */
+			ping_args.timeout = atoi(optarg);
+			if (ping_args.timeout == 0) {
+				shell_warn(
+					shell,
+					"timeout not an integer (> 0), defaulting to %d msecs",
+					ICMP_PARAM_TIMEOUT_DEFAULT);
+				ping_args.timeout = ICMP_PARAM_TIMEOUT_DEFAULT;
+			}
+			break;
+		case 'I': /* PDN CID */
+			ping_args.cid = atoi(optarg);
+			if (ping_args.cid == 0) {
+				shell_warn(
+					shell,
+					"CID not an integer (> 0), default context used");
+				ping_args.cid = MOSH_ARG_NOT_SET;
+			}
+			break;
+		case 'c': /* count */
+			ping_args.count = atoi(optarg);
+			if (ping_args.count == 0) {
+				shell_warn(
+					shell,
+					"count not an integer (> 0), defaulting to %d",
+					ICMP_PARAM_COUNT_DEFAULT);
+				ping_args.timeout = ICMP_PARAM_COUNT_DEFAULT;
+			}
+			break;
+		case 'i': /* interval */
+			ping_args.interval = atoi(optarg);
+			if (ping_args.interval == 0) {
+				shell_warn(
+					shell,
+					"interval not an integer (> 0), defaulting to %d",
+					ICMP_PARAM_INTERVAL_DEFAULT);
+				ping_args.interval =
+					ICMP_PARAM_INTERVAL_DEFAULT;
+			}
+			break;
+		case 'l': /* payload length */
+			ping_args.len = atoi(optarg);
+			if (ping_args.len > ICMP_IPV4_MAX_LEN) {
+				shell_error(
+					shell,
+					"Payload size exceeds the ultimate max limit %d",
+					ICMP_IPV4_MAX_LEN);
+				goto show_usage;
+			}
+			break;
+		case '6': /* force ipv6 */
+			ping_args.force_ipv6 = true;
+			break;
+		case 'h': /* help */
+		default:
+			goto show_usage;
+		}
+	}
+
+	/* Check that all mandatory args were given: */
+	if (ping_args.target_name == NULL) {
+		shell_error(shell, "-d destination, MUST be given. See usage:");
+		goto show_usage;
+	} else {
+		/* All good for args, get the current connection info and start the ping: */
+		int ret = 0;
+		struct pdp_context_info_array_t pdp_context_info_tbl;
+
+		ret = ltelc_api_pdp_contexts_read(&pdp_context_info_tbl);
+		if (ret) {
+			shell_error(shell,
+				    "cannot read current connection info: %d",
+				    ret);
+			goto show_usage;
+		} else {
```

Comment:
You could omit this else to reduce the indentation of the scope below, which is a bit hard to read.

Reply:
True, there was another one few lines above. Removed both else-statements to reduce indentation level and code got a lot cleaner overall.